Switch statements

ZThatWhiteGuyZ opened the pull request:
Made one question and answer for switch statements and would accept some feedback. Also changed the numbers on the .md file for Structs practice and answers because I was using them for reference. Probably just needs a merge on those two.

Iantho1999 committed:
Added page for Switch Statements

Commit:
Number updates to stucts practice and answers.

Replaced the numbered list for questions with @.

Commit:
Started Switch Statement Practice and Answers

Began writing switch practice questions and answers.

jerrettl commented on src/intro-to-c/switch-statements/switch-statements-practice-answers.md:
```diff
@@ -0,0 +1,61 @@
+
+Here is some additional practice with switch statements. 
+
+@. You're a time traveler with a computer, but it happens to have very limited capabilities thanks to the time police. You also happen to be a history buff that loves the Roman Empire, so you traveled back in time with your silly little programable computer to the silly little Roman Empire. You've decided to teach the Romans modern arabic numbers rather than their weird letter numbers. To expidit ...
```
I like this and think it's really funny, but I would take out the time police part, *only* because it's not mentioned anywhere else in the problem. If you mention the time police again in another sentence, it doesn't become just extra information.

``` diff
- @. You're a time traveler with a computer, but it happens to have very limited capabilities thanks to the time police.
+ @. You're a time traveler with a computer.
```

ZThatWhiteGuyZ replied:
Sounds to me like it's time to squeeze those police in somewhere else, so long as it doesn't convolute the question statement.

jerrettl commented on src/intro-to-c/switch-statements/switch-statements-practice-answers.md:
```diff
@@ -0,0 +1,61 @@
+{
+	int number;
+
+	do printf("Press 111 to exit.");
+	while(number != 111)
+	{
+		printf("What number would you like to conver?");
```
Just changing some slight syntax and a typo here. Do-while should have do at the beginning, while at the end. Then I moved the first print statement into the loop. You are free to put it somewhere else if you want.

``` diff
- 	do printf("Press 111 to exit.");
+ 	do
- 	while(number != 111)
	{
+			printf("Press 111 to exit.");
- 		printf("What number would you like to conver?");
+ 		printf("What number would you like to convert?");
```

ZThatWhiteGuyZ replied:
It has been quite a while since I've coded in C, so I may goof here and there.

jerrettl commented on src/intro-to-c/switch-statements/switch-statements-practice-answers.md:
```diff
@@ -0,0 +1,61 @@
+				printf("That was not a valid entry. Try again with a number between 1 and 10, or the exit value.");
+				break;
+		}
+	}
```
Moving the while to the end.

``` diff
- 		}
+ 		} while (number != 111);
```

jerrettl commented on src/intro-to-c/switch-statements/switch-statements-practice-answers.md:
```diff
@@ -0,0 +1,61 @@
+# Switch Statements
+
```
Completely up to you, but going anonymous author for this one?

ZThatWhiteGuyZ replied:
lmao I forgot to put my name.

jerrettl reviewed (commented):
I really like the goofy example, it makes everything better. I left a few syntax things here and there, check it out and let me know what you think. I'll see if I can get the merge in there with the numbering and all that.

jerrettl committed:
Merge branch 'main' into switch-statements

Commit:
Second Question and Answer + Small Updates

Added a new question and answer. Was able to work the time police back in.
Updates:
Cleared out any extra white space on the code blocks.  Updated the heading and added myself as an author to both pages.

Commit:
Question and Title Updates

Updated second question with some more information. Corrected syntax errors from previous commits.

Commit:
Question Updates

Further developed the story of question 1, and wrote the name of the next two questions I want to make.

Commit:
Added Question 3 + Answer

Continued the story of the problem set. This question focuses on code parsing and some tricky things with switch cases. considering adding a note about the null character, as it is present as an ascii in the final case.

Commit:
Added Question and Answer 4, Updated Question Text

Made Question 4, which deals with nested switches. Text updates include newlines in a handful of places to make the page format look correct.  Also cleared out some whitespace.

Commit:
Update to Text and Questions 3 and 4

Cleared trailing whitespace and corrected spelling errors. Added note about null character in question three. Updated list of inputs and put answer in code block for question/answer 3. Adjusted the code of answer 4 to be more easily readable.

Commit:
Update switch-statements-practice.md

Null put in code syntax . Forgot to save before pushing. Woops.

jerrettl committed:
switch-statements-practice-answers: Fix typo

jerrettl commented:
Merged.
